packages/@ama-sdk/core/src/fwk/api.helpers.ts

```diff
@@ -5,35 +5,200 @@ import type {
   ReviverType,
 } from './reviver';
 
+/** OpenAPI parameter serialization syntax */
+export interface ParamSerialization {
+  /**
+   * Value of the parameter
+   * Parameters can be primitive values, arrays, or objects
+   */
+  value: any;
+  /** Parameter is of exploded syntax */
+  exploded: boolean;
+  /** Style of the parameter */
+  style: string;
+}
+
+/**
+ * Check if the value is a record of type <string, string>.
+ * @param obj
+ */
+export function isRecordTypeString(obj: any): obj is { [key: string]: string } {
+  return Object.values(obj).every((item) => typeof item === 'string');
```

Comment:
This is `true` for any `type obj !== 'object'` maybe you should check:
```suggestion
  return  typeof obj === 'object' && Object.values(obj).every((item) => typeof item === 'string');
```